src/Facility.Definition/ServiceFieldValidationRange.cs

```diff
@@ -0,0 +1,83 @@
+using System;
+
+namespace Facility.Definition
+{
+	/// <summary>
+	/// A range for validating numeric values
+	/// </summary>
+	public sealed class ServiceFieldValidationRange
+	{
+		/// <summary>
+		/// Creates a range for validation
+		/// </summary>
+		/// <param name="startInclusive">The inclusive start of the range</param>
+		/// <param name="endInclusive">The inclusive end of the range</param>
+		public ServiceFieldValidationRange(decimal? startInclusive, decimal? endInclusive)
+		{
+			if (startInclusive == null && endInclusive == null)
+				throw new ArgumentException($"{nameof(startInclusive)} or {nameof(endInclusive)} must be specified");
+
+			StartInclusive = startInclusive;
+			EndInclusive = endInclusive;
+		}
```

Comment:
AFAICT, the major downside of always inclusive ranges means that it makes expressing some ranges (like negative numbers) hard since `..0` would include 0. The non-range syntax would suffer this problem also unless we had inclusive/exclusive options for each boundary (which we could do with a range also). Probably doesn't have to be a problem for now. I just thought it worth mentioning.

Reply:
I was thinking that wouldn't work for floating point, but maybe that's pretty contrived.

Reply:
Perhaps we can allow optional syntax for exclusive ranges in the future, e.g. `[0..1)` or `0..<1` or who knows what. But I think we can keep it simple for now. We can always respond to demand later.